### ISSUE TITLE
ALAsset in CDVURLProtocol.m is deprecated since iOS 9.0

### ISSUE THREAD

**Issue by maedewiza** [apache/cordova-ios#551]
# Bug Report

### What is expected to happen?
Cordova what uses PHAsset from the Photos framework instead of deprecated ALAsset.

### What does actually happen?
Cordova target is iOS 10, but in CDVURLProtocol.m uses deprecated API from before iOS 9.0.

## Information

This causes XCode warnings, but code to refactoring is a few lines.

### Command or Code

https://github.com/apache/cordova-ios/blob/master/CordovaLib/Classes/Public/CDVURLProtocol.m
This is deprecated code:
```
    if ([[url absoluteString] hasPrefix:kCDVAssetsLibraryPrefixes]) {
        ALAssetsLibraryAssetForURLResultBlock resultBlock = ^(ALAsset* asset) {
            if (asset) {
                // We have the asset!  Get the data and send it along.
                ALAssetRepresentation* assetRepresentation = [asset defaultRepresentation];
                NSString* MIMEType = (__bridge_transfer NSString*)UTTypeCopyPreferredTagWithClass((__bridge CFStringRef)[assetRepresentation UTI], kUTTagClassMIMEType);
                Byte* buffer = (Byte*)malloc((unsigned long)[assetRepresentation size]);
                NSUInteger bufferSize = [assetRepresentation getBytes:buffer fromOffset:0.0 length:(NSUInteger)[assetRepresentation size] error:nil];
                NSData* data = [NSData dataWithBytesNoCopy:buffer length:bufferSize freeWhenDone:YES];
                [self sendResponseWithResponseCode:200 data:data mimeType:MIMEType];
            } else {
                // Retrieving the asset failed for some reason.  Send an error.
                [self sendResponseWithResponseCode:404 data:nil mimeType:nil];
            }
        };
        ALAssetsLibraryAccessFailureBlock failureBlock = ^(NSError* error) {
            // Retrieving the asset failed for some reason.  Send an error.
            [self sendResponseWithResponseCode:401 data:nil mimeType:nil];
        };

        ALAssetsLibrary* assetsLibrary = [[ALAssetsLibrary alloc] init];
        [assetsLibrary assetForURL:url resultBlock:resultBlock failureBlock:failureBlock];
        return;
    }
```

### Environment, Platform, Device

iOS

### Version information

5.1

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

**Comment by brodybits** [comment]
Confirmed as a bug (see <https://developer.apple.com/documentation/assetslibrary/alasset>), thanks!

A pull request would definitely be welcome.

**Comment by brodybits** [comment]
Seems to be resolved by PR #783
